apps/src/lib/client/utils.rs

```diff
@@ -1142,16 +1170,26 @@ fn network_configs_url_prefix(chain_id: &ChainId) -> String {
 
 fn write_tendermint_node_key(
     tm_home_dir: &Path,
-    node_sk: ed25519::SecretKey,
-) -> ed25519::PublicKey {
-    let node_pk: ed25519::PublicKey = node_sk.ref_to();
-    // Convert and write the keypair into Tendermint
-    // node_key.json file
-    let node_keypair =
-        [node_sk.try_to_vec().unwrap(), node_pk.try_to_vec().unwrap()].concat();
+    node_sk: common::SecretKey,
+) -> common::PublicKey {
+    let node_pk: common::PublicKey = node_sk.ref_to();
+
+    // Convert and write the keypair into Tendermint node_key.json file.
+    // Tendermint requires concatenating the private-public keys for ed25519
+    // but does not for secp256k1.
+    let (node_keypair, key_str) = match node_sk {
+        common::SecretKey::Ed25519(_) => {
+            ([node_sk.try_to_vec().unwrap(), node_pk.try_to_vec().unwrap()].concat(),
```

Comment:
I think `node_sk.try_to_vec()` would produce a vector with a prefix byte indicating the type of key which Tendermint would not accept. Maybe something along the following lines would be better:
```
    let (node_keypair, key_str) = match node_sk {
        common::SecretKey::Ed25519(node_sk) => {
            ([node_sk.try_to_vec().unwrap(), node_sk.ref_to().try_to_vec().unwrap()].concat(),
            "Ed25519")
        },
        common::SecretKey::Secp256k1(node_sk) => {
            (node_sk.try_to_vec().unwrap(), "Secp256k1")
        },
    };
```

Reply:
Do you mean `node_pk.try_to_vec()`? The concatenation line you suggested concatenates `node_sk` to itself. Just wondering if the `node_sk` or `node_pk` should have a `.ref_to()`.

Reply:
Sorry, I keep missing the mistake that I made in the concatenation line... I guess the first part `node_sk.try_to_vec().unwrap()` represents the  byte encoding of an `ed25519::SecretKey` and the second part `node_sk.ref_to().try_to_vec().unwrap()` represents the byte encoding of `node_sk.ref_to()` which I guess is an `ed25519::PublicKey`. No?

Reply:
Ahh ok sorry, I understand what is happening here now in your suggestion. Yes, you are correct, I'll try to check if what you say about `node_sk.try_to_vec()` having an prefix byte from `common` is true. Thx for the suggestion